lib/cfa/placer.rb

```diff
@@ -11,14 +11,20 @@ def new_element(_tree)
       raise NotImplementedError,
         "Subclasses of #{Module.nesting.first} must override #{__method__}"
     end
+
+  protected
+
+    def create_element
+      { operation: :add }
+    end
   end
 
   # Places the new element at the end of the tree.
   class AppendPlacer < Placer
     # (see Placer#new_element)
     def new_element(tree)
-      res = {}
-      tree.data << res
+      res = create_element
+      tree.data(filtered: false) << res
```

Comment:
This `data(filtered: false)` everywhere is crazy. How about creating an *adapter tree* whose `data` does not have an argument and calls `@delegate.data(filtered: false)` ?